### PR TITLE
Properly use pmac device in plans

### LINKED ISSUE #98: I24 ssx: use PMAC device in plans
- Use PMAC device to move the motors with bluesky plans
- Send pmac_strings using `yield from abs_set`
- Tidy up PMAC device

### PR DESCRIPTION
**Pull request by noemifrisina**
Closes #98 

Needs [Dodal#561](https://github.com/DiamondLightSource/dodal/pull/561)

### CODE REVIEW

File: src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Collect_py3v1.py [hunk 1]
```diff
@@ -536,6 +538,12 @@ def finish_i24(
     return end_time
 
 
+def run_aborted_plan(pmac: PMAC):
```
Comment (DominicOram) [inline src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Collect_py3v1.py:541]: Should: What does `aborted_plan` mean in this context? Can you add a docstring?

File: src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Collect_py3v1.py [hunk 2]
```diff
@@ -601,7 +609,7 @@ def main():
 
     logger.info(f"Run PMAC with program number {prog_num}")
     logger.debug(f"pmac str = &2b{prog_num}r")
```
Comment (DominicOram) [inline src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Collect_py3v1.py:611]: Nit: `ophyd_async` will now log this for you and so you can remove

File: src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py [hunk 3]
```diff
@@ -49,29 +51,32 @@ def setup_logging():
...
+        pmac = i24.pmac()
+    group = "initialise_stages"
     # commented out filter lines 230719 as this stage not connected
     logger.info("Setting VMAX VELO ACCL HHL LLM pvs for stages")
 
+    # NOTE .VMAX is read only in ohpyd_async motor.
     caput(pv.me14e_stage_x + ".VMAX", 20)
     caput(pv.me14e_stage_y + ".VMAX", 20)
     caput(pv.me14e_stage_z + ".VMAX", 20)
     # caput(pv.me14e_filter  + '.VMAX', 20)
```
Comment (DominicOram) [inline src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py:65]: Should: I think it's correct that `VMAX` is read only. `VMAX` is the top speed that the motor has been commissioned with and so shouldn't be changed in normal operation. Can you discuss with controls/scientists with a view to removing setting the `VMAX` here at all

File: src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py [hunk 4]
```diff
@@ -82,10 +87,10 @@ def initialise():
...
     caput(pv.me14e_dcdetdist, 1480)
     caput(pv.me14e_exptime, 0.01)
-    caput(pv.me14e_pmac_str, "m508=100 m509=150")
-    caput(pv.me14e_pmac_str, "m608=100 m609=150")
-    caput(pv.me14e_pmac_str, "m708=100 m709=150")
-    caput(pv.me14e_pmac_str, "m808=100 m809=150")
+    yield from bps.abs_set(pmac.pmac_string, "m508=100 m509=150", group=group)
+    yield from bps.abs_set(pmac.pmac_string, "m608=100 m609=150", group=group)
+    yield from bps.abs_set(pmac.pmac_string, "m708=100 m709=150", group=group)
+    yield from bps.abs_set(pmac.pmac_string, "m808=100 m809=150", group=group)
```
Comment (DominicOram) [inline src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py:93]: Should: What are these commands doing? Can we either comment or better yet pull the settings into a nicely named constant?

File: src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py [hunk 5]
```diff
@@ -920,13 +926,17 @@ def cs_reset(pmac: PMAC = None):
...
     print(strg)
-    pmac.pmac_string.set("&2").wait()
-    pmac.pmac_string.set(cs1).wait()
-    pmac.pmac_string.set(cs2).wait()
-    pmac.pmac_string.set(cs3).wait()
+    yield from set_pmac_strings_for_cs(pmac, {"cs1": cs1, "cs2": cs2, "cs3": cs3})
     logger.debug("CSreset Done")
 
 
+def set_pmac_strings_for_cs(pmac: PMAC, cs_str: dict):
```
Comment (DominicOram) [inline src/mx_bluesky/I24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py:933]: Should: Can we have a docstring here about what this means?

**Review by DominicOram — DISMISSED** [review]
Great, thanks. Some comments in code. Additionally, we have the x, y, z epics motors, the co-ordinates we move by sending `#1` and the co-ordinates we move by sending `!x...`. Do we have some documentation on how these all relate to to each other, if not can we please write some or make a new issue to write some?

**Comment by noemifrisina** [comment]
> Great, thanks. Some comments in code. Additionally, we have the x, y, z epics motors, the co-ordinates we move by sending `#1` and the co-ordinates we move by sending `!x...`. Do we have some documentation on how these all relate to to each other, if not can we please write some or make a new issue to write some?

I think there might be something buried in confluence, but it would probably be a good idea to collate everything here - maybe in the wiki? 
Might take a bit of time to find everything/ask for clarifications, so opened a separate issue here #108

**Review by dperl-dls — APPROVED** [review]
looks good, changes were addressed

**Pull request merged**